src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3123,22 +3123,54 @@ void ViewProviderSketch::drawEditMarkers(const std::vector<Base::Vector2d>& Edit
 }
 
 void ViewProviderSketch::updateData(const App::Property* prop) {
-    if (std::string(prop->getName()) != "ShapeMaterial") {
-        // We don't want material to override the colors of sketches.
-        ViewProvider2DObject::updateData(prop);
-    }
-
```

Comment:
Why did you moved this to the end of the function? I have not studied the case, but this is not a trivial change and could have consequences.